src/app/public/a11y/a11y-analyzer.ts

```diff
@@ -1,113 +1,73 @@
-      'aria-required-attr': { enabled: true },
-      'aria-required-children': { enabled: true },
-      'aria-required-parent': { enabled: true },
-      'aria-roles': { enabled: true },
-      'aria-valid-attr': { enabled: true },
-      'aria-valid-attr-value': { enabled: true },
-      'audio-caption': { enabled: true },
-      'blink': { enabled: true },
-      'button-name': { enabled: true },
-      'bypass': { enabled: true },
-      'checkboxgroup': { enabled: true },
-      'color-contrast': { enabled: true },
-      'definition-list': { enabled: true },
-      'dlitem': { enabled: true },
-      'document-title': { enabled: true },
-      'duplicate-id': { enabled: true },
-      'empty-heading': { enabled: true },
-      'frame-title': { enabled: true },
-      'frame-title-unique': { enabled: true },
-      'heading-order': { enabled: true },
-      'html-has-lang': { enabled: true },
-      'html-lang-valid': { enabled: true },
-      'href-no-hash': { enabled: true },
-      'image-alt': { enabled: true },
-      'image-redundant-alt': { enabled: true },
-      'input-image-alt': { enabled: true },
-      'label': { enabled: true },
-      'layout-table': { enabled: true },
-      'link-in-text-block': { enabled: true },
-      'link-name': { enabled: true },
-      'list': { enabled: true },
-      'listitem': { enabled: true },
-      'marquee': { enabled: true },
-      'meta-refresh': { enabled: true },
-      'meta-viewport': { enabled: true },
-      'meta-viewport-large': { enabled: true },
-      'object-alt': { enabled: true },
-      'p-as-heading': { enabled: true },
-      'radiogroup': { enabled: true },
-      'scope-attr-valid': { enabled: true },
-      'server-side-image-map': { enabled: true },
-      'tabindex': { enabled: true },
-      'table-duplicate-name': { enabled: true },
-      'table-fake-caption': { enabled: true },
-      'td-has-header': { enabled: true },
-      'td-headers-attr': { enabled: true },
-      'th-has-data-cells': { enabled: true },
-      'valid-lang': { enabled: true },
-      'video-caption': { enabled: true },
-      'video-description': { enabled: true }
-    }
-  };
-
-  // Istanbul has difficulty covering abstract classes.
-  /*istanbul ignore next */
-  constructor() { }
-
-  public static run(element?: any, config?: SkyA11yAnalyzerConfig): Promise<any> {
-    const settings = Object.assign({}, SkyA11yAnalyzer.defaults, config);
-
-    return new Promise((resolve: Function, reject: Function) => {
-      SkyA11yAnalyzer.analyzer.run(element, settings, (error: Error, results: any) => {
+
+  private static analyzer = axe;
+
+  public static run(
+    element?: axe.ElementContext,
+    config?: SkyA11yAnalyzerConfig
+  ): Promise<void> {
+
+    const defaults: SkyA11yAnalyzerConfig = {
+      rules: { }
+    };
+
+    // Enable all rules by default.
+    axe.getRules().forEach((rule) => {
+      defaults.rules[rule.ruleId] = { enabled: true };
```

Comment:
Would we expect some SPAs to start having failing tests once this is turned on?

Reply:
Yep! Upgrading `axe-core` to the latest version introduces breaking changes, which is why this will need to be added to the RC branch.